### ISSUE TITLE
css line-clamp not compliant with SC reflow, text-spacing ?

### ISSUE THREAD

**Issue by goetsu** [w3c/wcag#1134]
Hello,

line-clamp or -webkit-line-clamp can be used to clamp text for design purpose when a text is longer that expected https://developer.mozilla.org/fr/docs/Web/CSS/-webkit-line-clamp

If we consider a block of text where text is always clamped in both desktop and mobile I would not consider it as a failure of reflow or text-spacing as it's the "default" for everyone in every context.

But what about when the clamping effect appear when 
* only on mobile for all text (all text clamped by default on mobile)
* only on mobile for some text where text longer that possible on mobile design or text spacing changed
* only on mobile for some text where text longer that possible on mobile design or text spacing but text is a link that go to a page where full text is available

**Comment by alastc** [comment]
Looking at the example in MDN, you loose content when applying text-spacing. Everyone looses some content, but you loose more when applying text spacing. If auditing a site, I'd be checking if that content is available nearby.

If by 'mobile' you mean small-screen variations, then it should be caught in regular testing.

If you mean using device detection for actual phones, then it would depend on your accessibility support and what would generally be available in those scenarios.

- only on mobile for all text: That probably does fit into your 'default for everyone' comment, if there is no mechanism to apply text-spacing.
- only on mobile for some text: As above.
- only on mobile for some text [and there] is a link that go to a page where full text is available: If the content is available nearby (including click-through) that could be considered an alternative conforming version.

**Comment by goetsu** [comment]
@alastc ok for link case or content available nearby
For case where everyone loose content as it's the intended design / feature for everyone in all situation even if you lost more when text-spacing increased or when reflow I would not consider this as as a failure because behaviours is the same for everyone and in my opinion it isn't a discrimination but a design decision (maybe a bad one but bad for everyone). 

Otherwise it also mean that line-clamp (unless mechanism provided to remove it or access full content) will always be not compliant (same will apply for text-overflow: ellipsis; property).

by mobile I mean small-screen (320px width to match testing case of reflow)

For me only the case where some texts are clamped because text spacing increased or reflow (but not all of them) should be considered a failure as some user will always be in such configuration

**Comment by patrickhlauke** [comment]
i'd say it will still be dependent on the particular thing that's clamped/elided. for instance, if it's an extract of a blog post/news article, showing a bit of a preview, but for all users they need to click through to read the full thing, you *could* argue that while yes, smaller viewports/increased text spacing users will get less of a preview, the point of that text is only to tease, and for all users they're expected to click through to read the full text.

so saying that it's outright "outlawed" is probably a bit too heavy-handed (though the presence of clamp/elipsis would make me look very closely at the purpose/utility of the thing)

**Comment by patrickhlauke** [comment]
and yes, saying that something just happens on mobile/small screen is of course not an excuse/a way to get around pass/fails, since https://www.w3.org/TR/WCAG21/#cc2 specifies that claims apply to full pages including their various variants based on screen size

**Comment by mraccess77** [comment]
I think this is more of an issue for low users who turn on large text settings such as on their iPhone and less text can fit.   My general feeling is that if the control could expand vertically then it should.  If the control say is a button and there is no room to expand then it becomes a less trivial issue to solve.   I do see some approaches like chat apps that allow you to expand the truncated text by tapping it.

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please do that and convert it to a Discussion.
